pkg/virt-controller/watch/vmi/storage.go

```diff
@@ -265,6 +265,51 @@ func (c *Controller) updateVolumeStatus(vmi *virtv1.VirtualMachineInstance, virt
 	return nil
 }
 
+func (c *Controller) checkEphemeralHotplugVolumes(vmi *virtv1.VirtualMachineInstance, vm *virtv1.VirtualMachine) {
+	vmVolumeMap := map[string]struct{}{}
+	if vmi == nil || vm == nil {
+		return
+	}
+
+	for _, volume := range vm.Spec.Template.Spec.Volumes {
+		vmVolumeMap[volume.Name] = struct{}{}
+	}
+
+	// account for volume requests that have not been updated in vm spec
+	for _, volume := range vm.Status.VolumeRequests {
+		if volume.AddVolumeOptions != nil {
+			vmVolumeMap[volume.AddVolumeOptions.Name] = struct{}{}
+		} else if volume.RemoveVolumeOptions != nil {
+			vmVolumeMap[volume.RemoveVolumeOptions.Name] = struct{}{}
```

Comment:
nbd but if there is an entry here the volume should be in `vm.Spec.Template.Spec.Volumes`